presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ScalarImplementation.java

```diff
@@ -419,15 +421,16 @@ else if (parameterType.isPrimitive()) {
 
                     boolean currentNullFlag = false;
                     if (i + 1 < method.getParameterCount()) {
-                        boolean foundIsNull = false;
                         annotations = method.getParameterAnnotations()[i + 1];
-                        for (Annotation annotation : annotations) {
-                            if (annotation instanceof IsNull) {
-                                foundIsNull = true;
-                                break;
+                        if (Stream.of(annotations).anyMatch(IsNull.class::isInstance)) {
+                            Class<?> isNullType = method.getParameterTypes()[i + 1];
+                            checkArgument(Stream.of(annotations).allMatch(IsNull.class::isInstance), "Method [%s] has @IsNull parameter that has other annotations", method);
```

Comment:
This check is overly broad. We should only restrict the specific annotations that I mentioned. The user might have a legitimate reason to a different (non-Presto) annotation.